docs/schemaform/walkthrough.md

```diff
@@ -0,0 +1,286 @@
+  items: {
+    type: 'object',
+    properties: {
+      field1: {
+        type: 'string'
+      },
+      field2: {
+        type: 'string'
+      }
+    }
+  }
+}
+```
+
+The field components pass a collection of props down through each component. The important ones are:
+
+- `name`
+  - The property name of the current field. Something like `field1`, if we were using the object schema above.
+- `required`
+  - If the field is required or not (i.e. the property name is in the schema's `required` array).
+- `schema`
+  - The schema for the specific field. 
+- `uiSchema`
+  - The ui schema for this field.
+- `errorSchema`
+  - This is an object that contains the list of errors for the current field and any child properties, if the field is an array or object.
+- `idSchema`
+  - This is an object that contains the field ids for the current field and any child properties. The library generates ids for each field by joining each property name with an underscore.
+- `formData`
+  - The actual data entered for the field so far.
+- `onChange`
+  - Function that's called when any data is changed
+- `onBlur`
+  - Function that's called when focus is lost on a widget
+
+These props are passed to all field components and most of them are passed to widget components.
+
+When a user enters data, each widget calls `onChange`, listed above. Each component in the hierarchy passes its own `onChange` handler to its child fields and when some child data changes, it combines it with the rest of its data and calls the `onChange` prop passed to it from its parent. So, for example, if we have this object schema:
+
+```
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    }
+  }
+}
+```
+
+If the user types 'a', the `TextWidget` for field1 will call `onChange` with 'a'. That `onChange` prop came from the parent `ObjectField` component, which will take `a` and put it in an object as `field1` (`{ field1: 'a' }`), then call the `onChange` prop it was passed. Once it reaches the top level `Form` component, rjsf will run the JSON Schema validation on it and pass the results back down through the component hierarchy.
+
+This is important to understand; similar to how Redux works, all state is kept in the root of the form, the `Form` component. Any data processing or validation happens in `Form` or is triggered by hooks provided by `Form`. The schemaform code built on top of this processes the schemas and form data in Redux, triggered by events provided by `Form`.
+
+### UI schema
+
+Along with the regular JSON Schema, a ui schema is also optionally defined for each field. This schema is used for ui specific options that don't fit within the JSON Schema standard. The primary use for it that's built into the library is for specifying custom fields and widgets for specific fields in the schema (using `ui:field` and `ui:widget`). We've extended it so that we also use it for label names, custom validation, and conditional required fields.
```

Comment:
"Conditionally required" is grammatically what we're looking for, methinks. Else, it'd be "conditional, required fields."

Mmm...English grammar. Yum.